100_essentials/greetings/README.md

```diff
@@ -0,0 +1,59 @@
+# Greetings
+
+Everything you need to complete this challenge can be found [here](https://docs.python.org/3.8/tutorial/controlflow.html#defining-functions).
+
+---
+
+Define a function named `greet` that takes in a name and a message as inputs and returns a greeting in the following format:
+
+```python
+>>> greet("Mark", "Oh, hi")
+'Oh, hi, Mark!'
+```
+
+Additionally, the message should default to "Hello" if none is provided.
+
+```python
+>>> greet("Dave")
+'Hello, Dave!'
+```
+
+Define a function named `greet_with_style` that takes a name as a required positional parameter, an arbitrary amount of extra positional parameters, an optional key word parameter named `msg` that defaults to 'Hello', another optional key word parameter named `sep` that defaults to ', ', and a required keyword parameter named `punctuation`. `greet_with_style` should return a greeting (name + sep + message + puncuation) with any functions passed as additional positional arguments applied to it.
+
+```python
+>>> def starify(s):
+...    return "*".join(list(s))
+...
+>>> def capitalize(s):
+...    return s.upper()
+...
+>>> greet_with_style("Citizen", starify, capitalize, msg="Greetings", punctuation="!", sep='*')
+'G*R*E*E*T*I*N*G*S***C*I*T*I*Z*E*N*!'
+```
+
+Define a function named `add_epitaphs` that will add epitaphs to a name. `add_epitaphs` should take a name and an arbitrary amount of key word arguments as inputs and return a string that is a name with all epitaphs added to it in the following format:
+
+```python
+>>> add_epitaphs("Gregson", Son="Greg", Leader="Kobolds")
+'Gregson, Son of Greg, Leader of Kobolds'
+
+>>> shout = lambda x: x.upper()
+>>> greet_with_style(add_epitaphs(
+...        "Gregson", Son="Greg", Leader="Kobolds"), shout, msg="Hello", punctuation="!")
+'HELLO, GREGSON, SON OF GREG, LEADER OF KOBOLDS!'
+```
```

Comment:
I think this relates to the comment I left on lines 21-32, we could essentially combine the two, drop the requirement that they build a totally separate function or accept functions as arguments (maybe move that to a bonus?) and teach them `*args` and `**kwargs` since this is a 100-level function challenge and we should likely focus on "here are ways you can build a single, flexible function with the tools Python gives you" before we dive too deep into the other stuff.